### ISSUE TITLE
No output from eslint

### ISSUE THREAD

**Issue by jdayton3** [vuejs/eslint-plugin-vue#478]
**Tell us about your environment**

* **ESLint Version: 4.19.1**
* **eslint-plugin-vue Version: 4.5.0**
* **Node Version: 8.9.3**
* **OS: Ubuntu 16.04**

**Please show your full configuration:**

`./.eslintrc.js`:
```js
module.exports = {
    "extends": "plugin:vue/recommended",
};
```

**What did you do? Please include the actual source code causing the issue.**

`./index.js`:
```js
// should report the no-dupe-keys error
export default {
    props: {
      foo: String
    },
    computed: {
      foo: {
        get () {}
      }
    },
    data: {
      foo: null
    },
    methods: {
      foo () {}
    }
  }
```

```bash
npm install eslint eslint-plugin-vue -g
eslint index.js
```

**What did you expect to happen?**

Some error should have been reported (this is the exact **incorrect** example from the [no-dupe-keys documentation](https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/no-dupe-keys.md)

**What actually happened? Please include the actual, raw output from ESLint.**

```txt
```

No output whatsoever. 

When running eslint with the `--debug` flag, I get the following:

```txt
  eslint:cli Running on files +0ms
  eslint:glob-util Creating list of files to process. +0ms
  eslint:ignored-paths Looking for ignore file in /home/jdayton3/Documents/cgi/test_vue +0ms
  eslint:ignored-paths Could not find ignore file in cwd +0ms
  eslint:cli-engine Processing /home/jdayton3/Documents/cgi/test_vue/index.js +0ms
  eslint:cli-engine Linting /home/jdayton3/Documents/cgi/test_vue/index.js +0ms
  eslint:config Constructing config file hierarchy for /home/jdayton3/Documents/cgi/test_vue +0ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading /home/jdayton3/Documents/cgi/test_vue/.eslintrc.js +2ms
  eslint:config-file Loading JS config file: /home/jdayton3/Documents/cgi/test_vue/.eslintrc.js +0ms
  eslint:config-file Loading plugin:vue/recommended +99ms
  eslint:config-file Attempting to resolve eslint-plugin-vue +0ms
  eslint:config-file Loading JS config file: /home/jdayton3/miniconda3/lib/node_modules/eslint-plugin-vue/lib/index.js +1ms
  eslint:config-file Loading /home/jdayton3/miniconda3/lib/node_modules/eslint-plugin-vue/lib/configs/strongly-recommended.js +73ms
  eslint:config-file Loading JS config file: /home/jdayton3/miniconda3/lib/node_modules/eslint-plugin-vue/lib/configs/strongly-recommended.js +1ms
  eslint:config-file Loading /home/jdayton3/miniconda3/lib/node_modules/eslint-plugin-vue/lib/configs/essential.js +1ms
  eslint:config-file Loading JS config file: /home/jdayton3/miniconda3/lib/node_modules/eslint-plugin-vue/lib/configs/essential.js +0ms
  eslint:config-file Loading /home/jdayton3/miniconda3/lib/node_modules/eslint-plugin-vue/lib/configs/base.js +2ms
  eslint:config-file Loading JS config file: /home/jdayton3/miniconda3/lib/node_modules/eslint-plugin-vue/lib/configs/base.js +0ms
  eslint:plugins Loaded plugin vue (eslint-plugin-vue@4.5.0) (from /home/jdayton3/miniconda3/lib/node_modules/eslint-plugin-vue/lib/index.js) +0ms
  eslint:config Using /home/jdayton3/Documents/cgi/test_vue/.eslintrc.js +185ms
  eslint:config-ops Using config from partial cache +0ms
  eslint:config-ops Apply environment settings to config +1ms
  eslint:config-ops Creating config for environment browser +0ms
  eslint:config-ops Creating config for environment es6 +1ms
  eslint:linter Linting code for /home/jdayton3/Documents/cgi/test_vue/index.js (pass 1) +0ms
  eslint:linter Generating fixed text for /home/jdayton3/Documents/cgi/test_vue/index.js (pass 1) +54ms
  eslint:text-fixer Applying fixes +0ms
  eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:cli-engine Linting complete in: 253ms +248ms
```

Thanks!

**Comment by michalsnik** [comment]
Hey @jdayton3 

The thing is we don't want to treat any javascript object as vue component as it's not always the case. You can read [here](https://github.com/vuejs/eslint-plugin-vue#attention) how we detect components, and in your case you should add comment above the object to tell eslint-plugin-vue that it's a component:

```js
// @vue/component
export default {
 ...
}
```

or rename the file to: `index.jsx`
